atc/worker/pool.go

```diff
@@ -281,6 +281,7 @@ func (pool *pool) SelectWorker(
 	labels := metric.StepsWaitingLabels{
 		Platform:   workerSpec.Platform,
 		TeamId:     strconv.Itoa(workerSpec.TeamID),
+		TeamName:   pool.provider.GetTeamByID(workerSpec.TeamID),
```

Comment:
When the team saw this PR during standup one concern that came up was "will this hit the db every time the metric is updated?". The db isn't being hit every time `Inc()` and `Dec()` are called, but it will get hit a lot more now (every single step that runs), even if Concourse isn't configured with any metric emitters. 

Is there a way to eliminate this db call completely? You'll notice that all the other fields on the struct are populated with information that's been passed into the function. Can we do the same for the `teamName`? Pass in the info we need?

Looking at what's being passed in already, the answer appears to be no. If we go up the call stack though, does someone else have this info?
Excluding tests, `SelectWorker()` is called in these places:
<img width="435" alt="image" src="https://user-images.githubusercontent.com/4583837/121744024-4421b480-cad0-11eb-8a08-bda490e65e25.png">

If we go to one of these functions, what info do we have at that level?

I found that each of these functions are part of a `Check/Task/Get/PutStep` struct which all contain a `StepMetadata` struct which contains the `TeamName`: https://github.com/concourse/concourse/blob/6841cd592dfe844599b73e333ea66c650f2f237b/atc/exec/step_metadata.go#L8-L20

Can we pass this in? Either the entire `StepMetadata` or the `TeamName`? This will avoid the need for any extra db calls.

Reply:
That's an interesting point. I agree it's not a good idea hit the db for that, especially for data that doesn't change so often over time.  I did a research in that direction, but I didn't find the origin myself, that's why I decide to hit the db. Thanks to point me to this.